redash/handlers/queries.py

```diff
@@ -50,7 +50,8 @@ def get(self):
 class QueryRecentAPI(BaseResource):
     @require_permission('view_query')
     def get(self):
-        recent = [d.to_dict() for d in models.Query.recent(current_user.id)]
+        recent = [d.to_dict() for d in models.Query.recent(
+                    limit_to_current_user=True)]
```

Comment:
In `Query.recent`, I always constraint query with `where(cls.access_groups.contains_any(*current_user.groups))`, no matter what value `limit_to_current_user`. So it will still be limited to those visible to current user.